client-sdks/client-js/src/types.ts

```diff
@@ -51,11 +62,17 @@ export interface GetAgentResponse {
 
 export type GenerateParams<T extends JSONSchema7 | ZodSchema | undefined = undefined> = {
   messages: string | string[] | CoreMessage[] | AiMessageType[];
-} & Partial<Omit<AgentGenerateOptions<T>, 'experimental_generateMessageId'>>;
+  output?: T;
+  experimental_output?: T;
+  runtimeContext?: RuntimeContext;
+} & WithoutMethods<Omit<AgentGenerateOptions<T>, 'output' | 'experimental_output' | 'runtimeContext'>>;
 
 export type StreamParams<T extends JSONSchema7 | ZodSchema | undefined = undefined> = {
   messages: string | string[] | CoreMessage[] | AiMessageType[];
-} & Omit<AgentStreamOptions<T>, 'onFinish' | 'onStepFinish' | 'experimental_generateMessageId'>;
+  output?: T;
+  experimental_output?: T;
+  runtimeContext?: RuntimeContext;
+} & WithoutMethods<Omit<AgentStreamOptions<T>, 'output' | 'experimental_output' | 'runtimeContext'>>;
```

Comment:
Since we're using `zodToJSONSchema` isn't it serializable? We do still ned to omit `experimental_generateMessageId` too

Reply:
Oh, or is that what your type does? Removes any input args that are functions?